garak/generators/huggingface.py

```diff
@@ -59,15 +58,20 @@ class Pipeline(Generator, HFCompatible):
     generator_family_name = "Hugging Face 🤗 pipeline"
     supports_multiple_generations = True
 
-    def _set_hf_context_len(self, config):
-        if hasattr(config, "n_ctx"):
-            if isinstance(config.n_ctx, int):
-                self.context_len = config.n_ctx
-
-    def __init__(self, name, do_sample=True, generations=10, device=0):
-        self.fullname, self.name = name, name.split("/")[-1]
+    def __init__(
+        self, name="", do_sample=True, generations=10, device=0, config_root=_config
```

Comment:
Correct it can now be set via config, I avoided impacts to constructors initially since tests do not exercise these fully yet.

I think this class in particular may be a good test bed for defining patterns on passing additional attributes that will be consumed or passed as configuration to things encapsulated in the class.  Your comments about `device_map` and other configuration values are candidates for getting good examples worked out.

Reply:
Agreed. This module is popular, and gets stretched the most (or at least, breaks the most, given the reporting on discord)